packages/webdriverio/src/commands/element/isDisplayed.js

```diff
@@ -34,20 +34,46 @@
  * </example>
  *
  * @alias element.isDisplayed
- * @return {Boolean} true if element(s)* [is|are] displayed
+ * @return {Boolean} true if element is displayed
  * @uses protocol/elements, protocol/elementIdDisplayed
  * @type state
  *
  */
 
+import { ELEMENT_KEY } from '../../constants'
+import { getBrowserObject } from '../../utils'
+import isElementDisplayedScript from '../../scripts/isElementDisplayed'
+
+const noW3CEndpoint = ['MicrosoftEdge', 'safari']
+
 export default async function isDisplayed() {
 
+    let browser = getBrowserObject(this)
+
     /*
      * This is only necessary as isDisplayed is on the exclusion list for the middleware
      */
     if (!this.elementId) {
         this.elementId = (await this.parent.$(this.selector)).elementId
     }
 
-    return this.elementId ? await this.isElementDisplayed(this.elementId) : false
+    /*
+     * if element was still not found it also is not displayed
+     */
+    if (!this.elementId) {
+        return false
+    }
+
+    /*
+     * https://www.w3.org/TR/webdriver/#element-displayedness
+     * Certain drivers have decided to remove the endpoint as the spec
+     * no longer dictates it. In those instances, we pass the element through a script
+     * that was provided by Brian Burg of safaridriver.
+     */
+    return browser.isW3C && noW3CEndpoint.includes(browser.capabilities.browserName) ?
```

Comment:
I don't think we need to the await's here since whatever calls isDisplayed will have to await on that anyways.

Reply:
Return await does a better job of error handling. I can pull up docs if you want (its why my middleware PR exposed those broken unit tests).

Reply:
OK, I'm fine with how it is then :)